src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5284,7 +5284,7 @@ private static bool CanReuseVariableDeclarator(CSharp.Syntax.VariableDeclaratorS
             TerminatorState saveTerm = _termState;
             bool isFixed = (flags & VariableFlags.Fixed) != 0;
             bool isConst = (flags & VariableFlags.Const) != 0;
-            bool isLocal = (flags & VariableFlags.Local) != 0;
+            bool isLocal = (flags & VariableFlags.LocalOrField) != 0;
```

Comment:
>bool isLocal [](http://example.com/codeflow?start=12&length=12)

Feels like this local should be renamed (`isLocalOrField`?)

Reply:
Renamed to `isLocalOrField`.

> Consider adding tests for scenarios below that are affected by this flag, such as a local initialized with a ref-returning lambda.

We have existing tests in `LambdaReturnTypeParsingTests`: `Ref_07`, `Ref_09`, and `Ref_11`. Let me know if there are scenarios that I've overlooked.

Reply:
Thanks. Those tests seem to be for locals though. A similar test for a field would be more relevant.

Reply:
Does `FieldDeclaration_Initializer_05` cover the scenario?

Reply:
Indeed. Thanks